fastcrawler/core/process.py

```diff
@@ -44,23 +44,26 @@ async def start(self, silent: bool = True) -> None:
         This method will disable scheduler temporarily to avoid duplicate running
         """
         if self.controller:
-            await self.controller.toggle_task(self.task.name, new_status=False)
+            await self.controller.toggle_task(str(self.task.name), new_status=False)
         await self.spider.start(silent=silent)
         if self.controller:
-            await self.controller.toggle_task(self.task.name, new_status=True)
+            await self.controller.toggle_task(str(self.task.name), new_status=True)
         return None
 
     async def stop(self) -> None:
         """Stop the crawling process"""
-        self.spider.is_stopped = True
```

Comment:
seems like you need a command pattern here

Reply:
do we? there's only 2 commands which has external APIs.

Reply:
even if you don't use the command pattern
I think stopping a spider should be a command not a variable